docs/guides/src/main/server/configuration.adoc

```diff
@@ -4,202 +4,237 @@
+<@kc.build parameters="--help"/>
 
-The initial admin user can be added manually using the web frontend when accessed from localhost or automatically using environment variables.
+.Example: Run the `build` command to set the database to PostgreSQL before startup:
+<@kc.build parameters="--db=postgres"/>
 
-To add the initial admin user using environment variables, set `KEYCLOAK_ADMIN` for the initial admin username and `KEYCLOAK_ADMIN_PASSWORD` for the initial admin password.
-Keycloak uses them at the first startup to create an initial user with administration rights.
-Once the first user with administrative rights exists, you can use the UI or the command line tool `kcadm.[sh|bat]` to create additional users.
+==== Second step: Start Keycloak using `--optimized`
+After a successful build, you can start Keycloak and turn off the default startup behavior by invoking the following command:
 
-== Using placeholders
+<@kc.build parameters="--optimized <configuration-options>"/>
 
-You are able to use placeholders to resolve the configuration option value from environment variables.
+The `--optimized` parameter tells Keycloak to assume a pre-built, already optimized Keycloak image is used. As a result, Keycloak avoids checking for and running a build directly at startup to save the time to walk through this process.
 
-.Using a placeholder to resolve the host from an environment variable
-[source, bash]
-----
-db-url-host=${r"${MY_DB_HOST}"}
-----
+You can invoke all configuration options at startup - these are all options in the <@links.server id="all-config"/> guide that are **not** marked with a tool icon.
 
-You are also able to default to a specific value if the environment variable can not be resolved:
+If a build option is found at startup with an equal value to the value used when invoking the `build`, it gets silently ignored when using the `--optimized` flag. If it has a different value than the value used when a build was invoked, a warning is shown in the logs and the previously built value is used. In order for this value to take effect, you have to run a new `build` before starting.
 
-.Setting a default value
-[source, bash]
-----
-db-url-host=${r"${MY_DB_HOST:mydb}"}
-----
+.Example: Create an optimized build, then start Keycloak using the --optimized parameter
+
+1. Set build option for postgres using the build command
+
+    <@kc.build parameters="--db=postgres"/>
 
-== Unsupported server options
+2. Set configuration options for postgres inside `conf/keycloak.conf`
 
-In most cases, the available options from the server configuration should suffice to configure the server.
-However, you might need to use properties directly from Quarkus to enable a specific behavior or capability that is missing in the keycloak configuration.
+    [source]
+    ----
+    db-url-host=keycloak-postgres
+    db-username=keycloak
+    db-password=change_me
+    hostname=mykeycloak.acme.com
+    https-certificate-file
+    ----
```

Comment:
Not sure why but it doesn't render correctly:
![Screenshot 2022-07-26 at 17 47 52](https://user-images.githubusercontent.com/13906855/181051466-7c6a1833-a57e-45fc-aa4d-851a958d0e33.png)
My guess is we can't use the indentation.

Reply:
changed, thanks for noticing